src/articles/ArticleBrowser.js

```diff
@@ -0,0 +1,256 @@
+import React, { useContext, useEffect, useState } from "react";
+import { APIContext } from "../contexts/APIContext";
+import useArticlePagination from "../hooks/useArticlePagination";
+import strings from "../i18n/definitions";
+import useShadowRef from "../hooks/useShadowRef";
+import LoadingAnimation from "../components/LoadingAnimation";
+import ArticleListBrowser from "./ArticleListBrowser";
+import ArticleSwipeBrowser from "./ArticleSwipeBrowser";
+import LocalStorage from "../assorted/LocalStorage";
+import useExtensionCommunication from "../hooks/useExtensionCommunication";
+import {setTitle} from "../assorted/setTitle";
+
+export default function ArticleBrowser({
+    articlesAndVideosList = [],
+    setArticlesAndVideosList  = () => {},
+    originalList = [],
+    setOriginalList = ([]) => {},
+    isSwipeView = false,
+    content,
+    searchQuery,
+    searchPublishPriority,
+    searchDifficultyPriority })
+{
+  const api = useContext(APIContext);
+
+  // UI and logic state
+  const [searchResultArticleList, setSearchResultArticleList] = useState([])
+  const [searchError, setSearchError] = useState(false);
+  const [reloadingSearchArticles, setReloadingSearchArticles] = useState(false);
+
+  const [isExtensionAvailable] = useExtensionCommunication();
+
+  // video toggle
+  // Next three vars required for the "Show Videos Only" toggle button
+  const [areVideosAvailable, setAreVideosAvailable] = useState(false);
+  const [isShowVideosOnlyEnabled, setIsShowVideosOnlyEnabled] = useState(false);
+  // Ref is needed since it's called in the updateOnPagination function. This function
+  // could have stale values if using the state constant.
+  const isShowVideosOnlyEnabledRef = useShadowRef(isShowVideosOnlyEnabled);
+
+  // modal pref
+  //The ternary operator below fix the problem with the getOpenArticleExternallyWithoutModal()
+  //getter that was outputting undefined string values when they should be false.
+  //This occurs before the user selects their own preferences.
+  //Additionally, the conditional statement needed to be tightened up due to JS's unstable behavior, which resulted
+  //in bool values changing on its own on refresh without any other external trigger or preferences change.
+  // A '=== "true"' clause has been added to the getters to achieve predictable and desired bool values.
+  const doNotShowRedirectionModal_LocalStorage = LocalStorage.getDoNotShowRedirectionModal() === "true";
+  const [doNotShowRedirectionModal_UserPreference, setDoNotShowRedirectionModal_UserPreference] = useState(
+    doNotShowRedirectionModal_LocalStorage,
+  );
+
+  // keep latest priorities for infinite scroll searchmore
+  const searchPublishPriorityRef = useShadowRef(searchPublishPriority);
+  const searchDifficultyPriorityRef = useShadowRef(searchDifficultyPriority);
+
+  // exclude hidden and saved article from the homepage
```

Comment:
this "defensive programming" where we're checking either for strings or for booleans makes the code ugly. Let's just trust that our objects have the right boolean in them, and if they don't then fix the problem at the source.